src/snowflake/snowpark/_internal/analyzer/metadata_utils.py

```diff
@@ -66,6 +67,16 @@ def infer_quoted_identifiers_from_expressions(
     return result
 
 
+def is_select_statement_child_metadata_same(
+    source_plan: "SelectStatement",
+) -> bool:
+    """
+    When source_plan doesn't have a projection, it's a simple `SELECT * from ...`,
+    which has the same metadata as it's child plan (source_plan.from_).
+    """
+    return source_plan.projection is None
```

Comment:
that seems a single line of check and only called at one place, do we really need a separate function for this?

Reply:
yeah let me remove this function